starter/src/main/java/com/starter/fullstack/dao/InventoryDAO.java

```diff
@@ -89,7 +91,11 @@ public Optional<Inventory> update(String id, Inventory inventory) {
    * @return Deleted Inventory.
    */
   public Optional<Inventory> delete(String id) {
-    // TODO
-    return Optional.empty();
+    /* getting the Inventory object from the database using its' ID */
+    Optional<Inventory> inventoryOpt = retrieve(id);
+    /* If the inventory was in the data base, then remove it. */
+    inventoryOpt.ifPresent(this.mongoTemplate::remove);
```

Comment:
try to do this in a single mongo call.
The fewer database calls you make, the more optimize the code will be